src/components/AvatarWrapper/AvatarWrapper.spec.js

```diff
@@ -1,76 +1,156 @@
-import { shallowMount } from '@vue/test-utils'
+import { mount, shallowMount } from '@vue/test-utils'
 import { cloneDeep } from 'lodash'
 import Vuex from 'vuex'
 
+import NcAvatar from '@nextcloud/vue/dist/Components/NcAvatar.js'
+
 import AvatarWrapper from './AvatarWrapper.vue'
 
 import storeConfig from '../../store/storeConfig.js'
 
 describe('AvatarWrapper.vue', () => {
 	let testStoreConfig
 	let store
+	const USER_ID = 'user-id'
+	const USER_NAME = 'John Doe'
+	const preloadedUserStatus = { status: 'online', message: null, icon: null }
 
 	beforeEach(() => {
 		testStoreConfig = cloneDeep(storeConfig)
-		// eslint-disable-next-line import/no-named-as-default-member
 		store = new Vuex.Store(testStoreConfig)
 	})
 
-	it('Renders user avatars properly', () => {
-		const wrapper = shallowMount(AvatarWrapper, {
-			store,
-			propsData: {
-				id: 'test-id',
-				source: 'users',
-				name: 'test-name',
-			},
+	describe('render user avatar', () => {
+		test('component renders NcAvatar with standard size by default', () => {
+			const wrapper = shallowMount(AvatarWrapper, {
+				store,
+				propsData: {
+					name: USER_NAME,
+				},
+			})
+
+			const avatar = wrapper.findComponent(NcAvatar)
+			expect(avatar.exists()).toBeTruthy()
+			expect(avatar.props('size')).toBe(44)
 		})
-		expect(wrapper.vm.iconClass).toBe('')
-		// Check that the first child is the avatar component
-		expect(wrapper.element.firstChild.nodeName).toBe('NCAVATAR-STUB')
-		expect(wrapper.props().size).toBe(32)
-	})
-	it('Renders group icons properly', () => {
-		const wrapper = shallowMount(AvatarWrapper, {
-			store,
-			propsData: {
-				id: '',
-				source: 'groups',
-				name: '',
-			},
+
+		test('component does not render NcAvatar for non-users', () => {
+			const wrapper = shallowMount(AvatarWrapper, {
+				store,
+				propsData: {
+					name: 'emails',
+					source: 'emails',
+				},
+			})
+
+			const avatar = wrapper.findComponent(NcAvatar)
+			expect(avatar.exists()).toBeFalsy()
+		})
+
+		test('component renders NcAvatar with smaller size', () => {
+			const wrapper = shallowMount(AvatarWrapper, {
+				store,
+				propsData: {
+					name: USER_NAME,
+					small: true,
+				},
+			})
+
+			const avatar = wrapper.findComponent(NcAvatar)
+			expect(avatar.props('size')).toBe(22)
+		})
+
+		test('component renders user name and status with accepted properties', async () => {
+			const wrapper = mount(AvatarWrapper, {
+				store,
+				propsData: {
+					id: USER_ID,
+					name: USER_NAME,
+					showUserStatus: true,
+					preloadedUserStatus,
+				},
+			})
+
+			const avatar = wrapper.findComponent(NcAvatar)
+			await avatar.vm.$nextTick()
+
+			expect(avatar.attributes('title')).toBe(USER_NAME)
+			expect(avatar.attributes('aria-label')).toContain(USER_NAME)
+			const status = wrapper.findComponent('.avatardiv__user-status--online')
+			expect(status.exists()).toBeTruthy()
```

Comment:
Setting attributes on the root element of `NcAvatar` is not a part of `AwatarWrapper` component. It's 3rdparty `NcAvatar` internal implementation.

We should test only that name is passed to NcAvatar. All the other is the NcAvatars's job and should be tested in NcAvatar.

`avatardiv__user-status--online` is also a very internal implementation of not a Talk component.

Reply:
replaced with prop checking